src/backend/vulkan/mod.rs

```diff
@@ -335,6 +335,7 @@ impl Instance {
         info!("Created new instance");
         info!("Enabled instance extensions: {:?}", inner.enabled_extensions);
 
+        #[allow(clippy::arc_with_non_send_sync)]
```

Comment:
I think this raises a question, do we want to mark the private inner types as send and Sync or add a bunch of allow attributes for these types of cases?

Reply:
I think in this case it would be reasonable to implement `Send` and `Sync` for the inner type. For the use in `EGLBufferReader::new` we can't do that since it's a `*mut wl_display` that it reference counts. (Though that could possibly be done in a different way.)

Reply:
```rust
 // SAFETY (Host Synchronization): InstanceInner is always stored in an Arc, therefore destruction is
 // synchronized (since the inner value of an Arc is always dropped on a single thread).
```

```rust
// SAFETY: Destruction is externally synchronized (using an internal Arc).
unsafe impl Send for Instance {}
unsafe impl Sync for Instance {}
```

I think this is safe, though it isn't really related to the fact an `Arc` is used. Without `Arc` you'd still only be able to destroy the object once, from one thread.

Reply:
Dropping an InstanceInner would be safe since it assumes it owns the underlying ash::Instance. Plus if it's Send and Sync the call to vkDestroyInstance is still externally synchronized (by the borrow checker). And ash makes that function unsafe, so you'd need to know the semantics of InstanceInner to not cause undefined behavior.

I think we can mark InstanceInner as Send + Sync and then remove the manual Send + Sync impls on Instance.